pkg/engine/jmespath/functions.go

```diff
@@ -713,13 +719,28 @@ func jpRegexReplaceAllLiteral(arguments []interface{}) (interface{}, error) {
 		return nil, formatError(invalidArgumentTypeError, regexReplaceAllLiteral, 3, "String or Real")
 	}
 
-	reg, err := regexp.Compile(regex.String())
+	reg, err := getOrCompileRegexp(regex.String())
 	if err != nil {
 		return nil, formatError(genericError, regexReplaceAllLiteral, err.Error())
 	}
 	return string(reg.ReplaceAllLiteral([]byte(src), []byte(repl))), nil
 }
 
+func getOrCompileRegexp(pattern string) (*regexp.Regexp, error) {
```

Comment:
This one is more debatable. Would you mind splitting this in two PRs ?

Reply:
Makes sense, removed this part from the PR.